src/CarterExtensions.cs

```diff
@@ -148,33 +155,27 @@ private static void ApplyGlobalBeforeHook(IApplicationBuilder builder, CarterOpt
         /// Adds Carter to the specified <see cref="IServiceCollection"/>.
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection"/> to add Carter to.</param>
-        /// <param name="assemblies">Optional array of <see cref="Assembly"/> to add to the services collection. If assemblies are not provided, Assembly.GetEntryAssembly is called.</param>
-        /// <param name="loggerFactory">Optional <see cref="ILoggerFactory"/> to be passed for tracing of Carter setup</param>
-        public static void AddCarter(this IServiceCollection services, ILoggerFactory loggerFactory)
-        {
-            var logger = loggerFactory.CreateLogger(typeof(CarterExtensions));
-            AddCarter(services, logger);
-        }
-
-        /// <summary>
-        /// Adds Carter to the specified <see cref="IServiceCollection"/>.
-        /// </summary>
-        /// <param name="services">The <see cref="IServiceCollection"/> to add Carter to.</param>
-        /// <param name="assemblies">Optional array of <see cref="Assembly"/> to add to the services collection. If assemblies are not provided, Assembly.GetEntryAssembly is called.</param>
-        /// <param name="logger">Optional <see cref="ILogger"/> to be passed for tracing of Carter setup</param>
-        public static void AddCarter(this IServiceCollection services, ILogger logger = null)
+        /// <param name="enableDiagnostics">Whether to enable diagnostics at startup.</param>
+        public static void AddCarter(this IServiceCollection services, bool enableDiagnostics = false)
```

Comment:
I wonder if we remove the `enableDiagnostics` and just rely on the LogLevel, therefore we always add the CarterDiagnostics to the IOC container and call `LogDiscoveredModules` but depending on the log level it will/ill not get logged. Thoughts?